### ISSUE TITLE
Show warning if use "name" as static class property

### ISSUE THREAD

**Issue by terbooter** [Microsoft/TypeScript#7932]
**Actual behavior:** 
```ts
class WSPacketEvent {
    public static name = "WSPacketEvent";

    constructor(public target:WS, public packet:InPacket) {

    }
}

function addListeners(ws:WS) {
        ws.on(WSPacketEvent.name, this.onPacket); 
       //Runtime error. 
       //TypeError: Cannot assign to read only property 'name' 
       //of function WSPacketEvent(target, packet) 
}
```
This is because of function's property "name" is read only in ES6
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name

If remove assignment
```ts
class WSPacketEvent {
    //public static name = "WSPacketEvent";

    constructor(public target:WS, public packet:InPacket) {

    }
}

function addListeners(ws:WS) {
        ws.on(WSPacketEvent.name, this.onPacket); 
       //Compile error. 
       //Error:(18, 29) TS2339: Property 'name' does not exist on type 'typeof WSPacketEvent'.
}
```
**Expected behavior:**
1) Show warning when try to write to class static property named "name"
2) Do not show error if try to read from class static property named "name", which was not explicitly declared in TS code

**TypeScript Version:** 
1.8.9   --target ES7

**Comment by dheeran1** [comment]
Same here..

**Comment by mhegazy** [comment]
same issue in https://github.com/Microsoft/TypeScript/issues/442
